e2e/app.spec.ts

```diff
@@ -28,27 +36,49 @@ describe('App', function() {
 
   let locations;
   let mapPage: MapPageObject;
+  let actionsListPage: ActionsListPageObject;
 
   beforeEach(async function() {
-    mapPage = new MapPageObject();
+    mapPage = new MapPageObject('map-page');
+    actionsListPage = new ActionsListPageObject('actions-list-page');
 
     // Insert 3 random locations into the database and sort them by ascending longitude and latitude.
-    locations = await Promise.all(new Array(3).fill(0).map(() => locationFixtures.location({ bbox: ONEX_BBOX })));
+    locations = await createData(3, locationFixtures.location, { bbox: ONEX_BBOX });
     locations.sort((a, b) => compareCoordinates(a.get('geometry'), b.get('geometry')));
 
+    // Insert 3 random actions into the database.
+    await createData(6, actionFixtures.action, {});
+
     // Set a window size with the same width/height ratio as the Onex bounding box.
     const windowWidth = 1440;
     const windowHeight = windowWidth / (ONEX_BBOX_WIDTH / ONEX_BBOX_HEIGHT);
     await mapPage.setWindowSize(windowWidth, windowHeight);
   });
 
-  it('should allow a user to view locations on the map', async function() {
-    this.timeout(10000);
+  /**
+   * --- Main e2e scenario ---
+   *
+   * This end-to-end scenario simulates a user going through the following steps :
+   * 1. Launching the BioPocket app
+   * 2. Arriving on the root page, which should be the Map Page
+   * 3. Clicking on a Location map marker, and thus displaying the popover with the correct information
+   * 4. Dismissing the popover by clicking on its backdrop
+   * 5. Navigating to the Actions List page by clicking on the adequate button on the Map page
+   * 6. Scrolling to the bottom of the actions list page and trigger a new load
+   */
+  it('should allow a user to execute the main scenario', async function() {
+    this.timeout(15000);
 
     // Navigate to the map page.
     await mapPage.navigateTo();
     await expect(mapPage.getTitle()).to.eventually.equal('BioPocket');
 
+    // Wait for the actions list page to show up on the DOM
+    const mapPageFinder = mapPage.getPage();
+    await presenceOf(mapPageFinder);
+    await expectDisplayed(mapPageFinder, 'MapPage is not displayed while it should be.');
```

Comment:
In future PR, I removed some calls to `expectDisplayed` because they were completely useless, especially when called after a `presenceOf`. I propose we keep those lines until the latest PR is published.

Reply:
Indeed.